sdcm/cluster_k8s/eks.py

```diff
@@ -214,29 +214,25 @@ def launch_template_name(self) -> str:
 
     @property
     def is_launch_template_required(self) -> bool:
-        return bool(self.user_data)
+        # NOTE: use launch template to be able to specify the `gp3` disk type by default
+        return True
 
     @property
     def _launch_template_cfg(self) -> dict:
-        block_devices = []
+        root_ebs_block_device_params = {
+            "DeleteOnTermination": True,
+            "Encrypted": False,
+            "VolumeType": self.disk_type,
+        }
         if self.disk_size:
-            root_disk_def = LaunchTemplateBlockDeviceMappingRequestTypeDef(
-                DeviceName='/dev/xvda',
-                Ebs=LaunchTemplateEbsBlockDeviceRequestTypeDef(
-                    DeleteOnTermination=True,
-                    Encrypted=False,
-                    VolumeSize=self.disk_size,
-                    VolumeType=self.disk_type
-                ))
-            block_devices.append(root_disk_def)
+            root_ebs_block_device_params["VolumeSize"] = self.disk_size
+        root_disk_def = LaunchTemplateBlockDeviceMappingRequestTypeDef(
+            DeviceName="/dev/xvda",
+            Ebs=LaunchTemplateEbsBlockDeviceRequestTypeDef(**root_ebs_block_device_params))
         launch_template = RequestLaunchTemplateDataTypeDef(
             KeyName=self.ssh_key_pair_name,
             EbsOptimized=False,
-            BlockDeviceMappings=block_devices,
-            NetworkInterfaces=[{
```

Comment:
This isn't needed ?

Reply:
It is prohibited here, because security groups are specified out of the launch-template already.

Reply:
So before this change we weren't using the template at all ?

Do we clean those templates ?

Reply:
Yes, we were not using `launch templates` before.
No, we do not clean those templates.

Reply:
we should add the needed code for cleaning them up, in a followup PR